maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/GoalRenderer.java

```diff
@@ -0,0 +1,526 @@
+                attributeKey,
+                Optional.ofNullable(conditionAndCodeArgument));
+    }
+
+    private void renderAttribute(boolean condition, String attributeKey, Optional<String> codeArgument) {
+        if (condition) {
+            sink.listItem();
+            linkPatternedText(getI18nString(attributeKey));
+            if (codeArgument.isPresent()) {
+                text(": ");
+                sink.inline(Semantics.CODE);
+                sink.text(codeArgument.get());
+                sink.inline_();
+            }
+            text(".");
+            sink.listItem_();
+        }
+    }
+
+    private void renderParameterOverviewTable(String title, Iterator<Parameter> parameters) {
+        // don't emit empty tables
+        if (!parameters.hasNext()) {
+            return;
+        }
+        startSection(title);
+        startTable();
+        tableHeader(new String[] {
+            getI18nString("parameter.name.header"),
+            getI18nString("parameter.type.header"),
+            getI18nString("parameter.since.header"),
+            getI18nString("parameter.description.header")
+        });
+        while (parameters.hasNext()) {
+            renderParameterOverviewTableRow(parameters.next());
+        }
+        endTable();
+        endSection();
+    }
+
+    private void renderTableCellWithCode(String text) {
+        renderTableCellWithCode(text, Optional.empty());
+    }
+
+    private void renderTableCellWithCode(String text, Optional<String> link) {
+        sink.tableCell();
+        if (link.isPresent()) {
+            sink.link(link.get(), null);
+        }
+        sink.inline(Semantics.CODE);
+        sink.text(text);
+        sink.inline_();
+        if (link.isPresent()) {
+            sink.link_();
+        }
+        sink.tableCell_();
+    }
+
+    private void renderParameterOverviewTableRow(Parameter parameter) {
+        sink.tableRow();
+        // name
+        // link to appropriate section
+        renderTableCellWithCode(
+                format("parameter.name", parameter.getName()),
+                // no need for additional URI encoding as it returns only URI safe characters
+                Optional.of("#" + HtmlTools.encodeId(parameter.getName())));
+
+        // type
+        Map.Entry<String, Optional<String>> type = getLinkedType(parameter, true);
+        renderTableCellWithCode(type.getKey(), type.getValue());
+
+        // since
+        String since = StringUtils.defaultIfEmpty(parameter.getSince(), "-");
+        renderTableCellWithCode(since);
+
+        // description
+        sink.tableCell();
+        String description;
+        String context = "Parameter " + parameter.getName() + " in goal " + descriptor.getGoal();
+        if (StringUtils.isNotEmpty(parameter.getDeprecated())) {
+            String deprecated = getXhtmlWithValidatedLinks(parameter.getDescription(), context);
+            description = format("parameter.deprecated", deprecated);
```

Comment:
@kwin @michael-o
Is it intended - we check parameter.getDeprecated() but use parameter.getDescription()?

We lost here deprecated reason ... only deprecated label with standard description is printed.

Reply:
That is a copy-paste mistake.

Reply:
The removed Xdoc generator had it right. @slawekjaranowski Please file an issue. We need to fix this.